torch/functional.py

```diff
@@ -324,6 +325,118 @@ def parse_subscript(n: int) -> str:
 
     return _VF.einsum(equation, operands)  # type: ignore[attr-defined]
 
+# If pad specifier is not None and not a tensor, convert it to a tensor
+def _maybe_convert_pad_specifier(pad_spec, arg_name, dtype, device):
+    if pad_spec is None or torch.is_tensor(pad_spec):
+        return pad_spec
+
+    # TODO: torch.tensor(), when called outside of pytorch, throws a warning
+    # if dtype is an integer type and one of the inputs is a float. It would
+    # be nice to use that feature here to detect that case fairly efficiently.
+    # However, for some reason, the warning is not thrown when we call
+    # torch.tensor() here. Something must be disabling the warning, and I'm
+    # not sure how to reenable it.
+    '''
+    import warnings
+
+    with warnings.catch_warnings(record=True) as w_list:
+        pad_spec_tensor = torch.tensor(pad_spec, device=device, dtype=dtype)
+
+    for w in w_list:
+        if isinstance(w.message, DeprecationWarning):
+            if 'an integer is required' in str(w):
+                raise RuntimeError(f"'{arg_name}' must be of integral type")
+
+    return pad_spec_tensor
+    '''
+
+    pad_spec_tensor = torch.tensor(pad_spec, device=device)
```

Comment:
This is interesting; it might be helpful to elaborate on the translation from the various pad width inputs to a tensor here. It's nice that it happens through just a call to `torch.tensor()`